### ISSUE TITLE
Incompatibility over Angular 15

### ISSUE THREAD

**Issue by pheetah** [scttcper/ngx-toastr#964]
After migrating to Angular 15, I had errors on ToastrInjector. Full Error message is like this:

```
./src/styles.scss - Error: Module build failed (from ./node_modules/css-loader/dist/cjs.js):
Error: Can't resolve '~ngx-toastr/toastr.css' in '/Users/XXXXXX'

./src/styles.scss?ngGlobalStyle - Error: Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
HookWebpackError: Module build failed (from ./node_modules/css-loader/dist/cjs.js):
Error: Can't resolve '~ngx-toastr/toastr.css' in '/Users/XXXXXX'

Error: node_modules/ngx-toastr/toastr/toast-injector.d.ts:47:5 - error TS2416: Property 'get' in type 'ToastInjector' is not assignable to the same property in base type 'Injector'.
  Type '<T>(token: any, notFoundValue?: T | undefined, flags?: InjectFlags | undefined) => T | ToastPackage' is not assignable to type '{ <T>(token: ProviderToken<T>, notFoundValue: undefined, options: InjectOptions & { optional?: false | undefined; }): T; <T>(token: ProviderToken<T>, notFoundValue: null | undefined, options: InjectOptions): T | null; <T>(token: ProviderToken<...>, notFoundValue?: T | undefined, options?: InjectOptions | ... 1 more ...'.
    Types of parameters 'flags' and 'options' are incompatible.
      Type 'InjectOptions & { optional?: false | undefined; }' is not assignable to type 'InjectFlags | undefined'.

47     get<T>(token: any, notFoundValue?: T, flags?: InjectFlags): T | ToastPackage;
```

I guess there were some changes in Angular's Injector which causes problem with implementations of ToastrInjector (it implements Angular's Injector interface)

```
export declare class ToastInjector implements Injector {
    private _toastPackage;
    private _parentInjector;
    constructor(_toastPackage: ToastPackage, _parentInjector: Injector);
    get<T>(token: any, notFoundValue?: T, flags?: InjectFlags): T | ToastPackage;
}
```

**_InjectFlags_** is deprecated in Angular version 15. New type description on _flags_ property needed, currently changing in my projects ` get<T>(token: any, notFoundValue?: T, flags?: any): T | ToastPackage;` as workaround.

### package.json
`
"ngx-toastr": "^14.3.0",`

**Comment by pheetah** [comment]
After implementing `get<T>(token: any, notFoundValue?: T, flags?: any): T | ToastPackage; `, didn't change toaster's behaviour, it works fine with Angular 15. It probably needs correct type implementation over **_flags_** property in the function declaration.

[Might be interesting to see](https://github.com/angular/angular/issues/47164)

**Comment by pheetah** [comment]
I closed the issue, since 14.3.0 is a way older version.
